onnxruntime/wasm/api.cc

```diff
@@ -387,3 +395,109 @@ char* OrtEndProfiling(ort_session_handle_t session) {
+}
+
+ort_tensor_handle_t EMSCRIPTEN_KEEPALIVE OrtTrainingEvalStep(ort_training_session_handle_t training_handle,
+                                                             const ort_run_options_handle_t options,
+                                                             size_t inputs_len,
+                                                             const ort_tensor_handle_t* inputs,
+                                                             size_t outputs_len,
+                                                             ort_tensor_handle_t* outputs) {
+  return (CHECK_TRAINING_STATUS(EvalStep, training_handle,
+                                options, inputs_len, inputs, outputs_len, outputs) == ORT_OK)
+             ? *outputs
+             : nullptr;
+}
+
+// void EMSCRIPTEN_KEEPALIVE OrtTrainingExportModelForInferencing(orttraining_session_handle_t session,
+//                                                                const ORTCHAR_T* inference_model_path,
+//                                                                size_t graph_outputs_len,
+//                                                                const char* const* graph_output_names) {
+//   Ort::GetTrainingApi().ExportModelForInferencing(session, inference_model_path, graph_outputs_len, graph_output_names);
+// }
+
+size_t* EMSCRIPTEN_KEEPALIVE OrtTrainingGetParametersSize(ort_training_session_handle_t training_handle,
+                                                          bool trainable_only) {
+  size_t* param_size = nullptr;
+  return (CHECK_TRAINING_STATUS(GetParametersSize, training_handle, param_size, trainable_only) == ORT_OK)
+             ? param_size
+             : nullptr;
+}
+
+ort_tensor_handle_t EMSCRIPTEN_KEEPALIVE OrtTrainingCopyParametersToBuffer(ort_training_session_handle_t training_handle,
+                                                                           ort_tensor_handle_t parameters_buffer,
+                                                                           bool trainable_only) {
+  return (CHECK_TRAINING_STATUS(CopyParametersToBuffer, training_handle,
+                                parameters_buffer, trainable_only) == ORT_OK)
+             ? parameters_buffer
+             : nullptr;
+}
+
+ort_tensor_handle_t EMSCRIPTEN_KEEPALIVE OrtTrainingCopyBufferToParameters(ort_training_session_handle_t training_handle,
+                                                                           ort_tensor_handle_t parameters_buffer,
+                                                                           bool trainable_only) {
+  return (CHECK_TRAINING_STATUS(CopyBufferToParameters, training_handle,
+                                parameters_buffer, trainable_only) == ORT_OK)
+             ? parameters_buffer
+             : nullptr;
+}
+
+void EMSCRIPTEN_KEEPALIVE OrtTrainingReleaseSession(ort_training_session_handle_t training_handle) {
+  Ort::GetTrainingApi().ReleaseTrainingSession(training_handle);
```

Comment:
why not use the CHECK_TRAINING_STATUS macro here?

Reply:
I followed the pattern for how inference handled releasing, but my understanding is that the ReleaseTrainingSession API methods are implemented differently & the ReleaseTrainingSession method doesn't return an OrtStatusPtr, which is used for the CHECK_STATUS macros

Reply:
`ReleaseTrainingSession()` return void.

I think most of the `ReleaseXXX()` functions in ORT API returns void. It is designed not to fail.